README.md

```diff
@@ -561,52 +619,85 @@ class MyBaseClass extends WidgetBase<WidgetProperties> {
 }
 ```
 
-And using the `beforeRender` function for non decorator environments:
+##### AfterRender
+
+The `afterRender` lifecycle hook receives the returned `DNode`s from a widget's `render` call, so that the nodes can be decorated, manipulated or even swapped out.
 
 ```ts
 class MyBaseClass extends WidgetBase<WidgetProperties> {
-    constructor() {
-        super();
-        beforeRender(this.myOtherBeforeRender)(this);
-    }
-
-    myOtherBeforeRender(renderFunc: () => DNode, properties: any, children: DNode[]): () => DNode {
+    @afterRender()
+    myAfterRender(result: DNode): DNode {
         // do something with the result
-        return renderFunc;
+        return result;
     }
 }
 ```
 
-##### AfterRender
+### Containers & Injectors
 
-The `afterRender` call receives the returned `DNode`s from a widget's `render` call, so that the nodes can decorated, manipulated or even swapped.
+There is built in support for side-loading/injecting values into sections of the widget tree and mapping them to a widget's properties. This is achieved by registering a `@dojo/widget-core/Injector` instance, against a `registry` that is available to your application (i.e. set on the projector instance, `projector.setProperties({ registry })`).
+
+Create an `Injector` instance and pass the `payload` that needs to be injected to the constructor:
 
 ```ts
-class MyBaseClass extends WidgetBase<WidgetProperties> {
-    @afterRender()
-    myAfterRender(result: DNode): DNode {
-        // do something with the result
-        return result;
-    }
+const injector = new Injector({ foo: 'baz' });
+registry.defineInjector('my-injector', injector);
+```
+
+To connect the registered `injector` to a widget we can use the `Container` HOC (higher order component) provided by `widget-core`. The `Container` accepts a widget `constructor`, `injector` label and `getProperties` mapping function as arguments and returns a new class that returns the passed widget from its `render` function.
+
+`getProperties` receives the `payload` from an `injector` and `properties` from the container HOC component, that are used to map into the wrapped widgets properties.
+
+```ts
+import { Container } from '@dojo/widget-core/Container';
+import { MyWidget } from './MyWidget';
+
+function getProperties(payload: any, properties: any) {
+    return {
+        foo: payload.foo
+    };
 }
+
+export const MyWidgetContainer = Container(MyWidget, 'my-injector', getProperties);
 ```
 
-And using the `afterRender` function for non decorator environments:
+The returned class from `Container` HOC is then be used in place widget it wraps, not although the container assumes the properties type of the wrapped widget, they all become optional.
```

Comment:
The returned class from `Container` HOC is then be used in place widget it wraps, not although the container assumes the properties type of the wrapped widget, they all become optional. -> The returned class from `Container` HOC is then used in place of the widget it wraps, although if the container assumes the properties type of the wrapped widget, they all become optional.  ? (last part isn't clear to me)